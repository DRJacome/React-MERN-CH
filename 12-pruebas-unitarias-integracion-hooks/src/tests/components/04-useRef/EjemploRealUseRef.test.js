import React from 'react'
import { shallow } from 'enzyme'
import { MultipleCustomHooks } from "../../../components/03-ejemplos/MultipleCustomHooks"
import { EjemploRealUseRef } from '../../../components/04-useRef/EjemploRealUseRef'

describe('Pruebas en <EjemploRealUseRef/>', () => {

  const wrapper = shallow(<EjemploRealUseRef />);

  test('Debe mostrarse correctamente', () => {

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(MultipleCustomHooks).exists()).toBe(false);
  })

  test('Debe mostrar el compontente <MultipleCustomHooks/>', () => {

    wrapper.find('button').simulate('click')

    expect(wrapper.find(MultipleCustomHooks).exists()).toBe(true);
  })
})