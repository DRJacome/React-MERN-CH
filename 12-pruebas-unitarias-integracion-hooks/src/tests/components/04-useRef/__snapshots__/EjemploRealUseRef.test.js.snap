// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <EjemploRealUseRef/> Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    UseRef
  </h1>
  <hr />
  <button
    className="btn btn-primary m-5"
    onClick={[Function]}
  >
    Mostrar/ocultar
  </button>
</div>
`;
